src/BudgetManager/BudgetManager.CQRS/Handlers/CategoryHandlers/GetCategoriesHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using AutoMapper;
+using BudgetManager.CQRS.Queries.CategoryQueries;
+using BudgetManager.CQRS.Responses.CategoryResponses;
+using BudgetManager.Model;
+using BudgetManager.Shared.DataAccess.MongoDB.BaseImplementation;
+using MediatR;
+
+namespace BudgetManager.CQRS.Handlers.CategoryHandlers
+{
+    public class GetCategoriesHandler : IRequestHandler<GetCategoriesQuery, List<CategoryResponse>>
+    {
+        private readonly IBaseRepository<Category> _categoryRepository;
+        private readonly IMapper _mapper;
+
+        public GetCategoriesHandler(IBaseRepository<Category> categoryRepository, IMapper mapper)
+        {
+            _categoryRepository = categoryRepository;
+            _mapper = mapper;
+        }
+
+        public async Task<List<CategoryResponse>> Handle(GetCategoriesQuery request, CancellationToken cancellationToken)
+        {
+            var allCategoriesFromDb = await _categoryRepository.GetAllAsync(cancellationToken);
+            var listOfResponseCategories = new List<CategoryResponse>();
+            foreach (var category in allCategoriesFromDb)
+            {
+                CategoryResponse mappedCategory = _mapper.Map<CategoryResponse>(category);
+                listOfResponseCategories.Add(mappedCategory);
+            }
+
+            return listOfResponseCategories;
```

Comment:
I think if we receive IEnumerable from repo we should also return it as IEnumerable.
Also this code could be shortened to one line of code.

return  _mapper.Map<IEnumerable<CategoryResponse>>(await _dataAccess.GetAllAsync(cancellationToken));

Reply:
Sorry for a sc. Github comments didn't want to accept this piece of text for some reason.